File: Move_Negative_Elements_to_OneSide/Move_Negative_Element_to_oneSide.java
public class Move_Negative_Element_to_oneSide {
    public static void main(String[] args) {
        int[] arr = { 12, -7, 5, -3, 9, -1, 0, -6, 8 };

        moveNegativesToOneSide(arr);

        System.out.print("Array after moving negative elements to one side: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
    }

    private static void moveNegativesToOneSide(int[] arr) {
        int left = 0;
        int right = arr.length - 1;

        while (left < right) {
            if (arr[left] < 0) {
                left++;
            } else if (arr[right] >= 0) {
                right--;
            } else {
                int temp = arr[left];
                arr[left] = arr[right];
                arr[right] = temp;
                left++;
                right--;
            }
        }
    }
}

File: Move_Negative_Elements_to_OneSide/Move_Negative_Element_to_oneSide.c
#include <stdio.h>

void moveNegatives(int arr[], int n) {
    int left = 0, right = n - 1;
    while (left < right) {
        while (arr[left] < 0 && left < right) left++;
        while (arr[right] >= 0 && left < right) right--;
        if (left < right) {
            int temp = arr[left];
            arr[left] = arr[right];
            arr[right] = temp;
            left++;
            right--;
        }
    }
}

int main() {
    int arr[] = {-1, 2, -3, 4, 5, -6};
    int n = sizeof(arr) / sizeof(arr[0]);
    moveNegatives(arr, n);
    for (int i = 0; i < n; i++) printf("%d ", arr[i]);
    printf("\n");
    return 0;
}